drivers/clock_control/renesas/r8a77951-cpg-mssr.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2022 IoT.bzh
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_r8a7795_cpg_mssr
+
+#include <errno.h>
+#include <soc.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/renesas-cpg-mssr.h>
+#include <dt-bindings/clock/renesas/renesas-cpg-mssr.h>
+#include <dt-bindings/clock/renesas/r8a7795-cpg-mssr.h>
+#include "renesas-cpg-mssr.h"
+
+struct r8a77951_mssr_config {
+	uint32_t base_address;
+};
+
+int r8a77951_cpg_core_clock_endisable(uint32_t base_address, uint32_t module,
+				      uint32_t rate, bool enable)
+{
+	uint32_t divider;
+	unsigned int key;
+	int ret = 0;
+
+	/* Only support CANFD core clock at the moment */
+	if (module != R8A7795_CLK_CANFD) {
+		return -EINVAL;
+	}
+
+	key = irq_lock();
+
+	if (enable) {
+		if ((CANFDCKCR_PARENT_CLK_RATE % rate) != 0) {
+			__ASSERT(true, "Can not generate "
+				 "%u from CANFD parent clock", rate);
+			ret = -EINVAL;
+			goto unlock;
+		}
+		divider = (CANFDCKCR_PARENT_CLK_RATE / rate) - 1;
```

Comment:
Missing guard against `rate == 0`.

Reply:
Added a guard against rate equal or less to 0.